packages/react/src/FieldSet/FieldSet.test.tsx

```diff
@@ -54,4 +54,36 @@ describe('FieldSet', () => {
 
     expect(ref.current).toBe(component)
   })
+
+  it('renders the provided hint text after the legend', () => {
+    render(<FieldSet legend="Legend" hint="hint text" />)
+
+    const component = screen.getByRole('group', { name: 'Legend (hint text)' })
+
+    expect(component).toBeInTheDocument()
+  })
+
+  it('renders the default hint text after the legend', () => {
+    render(<FieldSet legend="Legend" optional />)
+
+    const component = screen.getByRole('group', { name: 'Legend (niet verplicht)' })
+
+    expect(component).toHaveTextContent('Legend (niet verplicht)')
```

Comment:
You can just use `expect(component).toBeInTheDocument()` here. If the text is incorrect, `component` will not exist.